File: Matlab MRI/t2pulseacquire.m
%{

This program plots the results
of a T2 pulse acquire on the real axis and imaginary axis,
along with the magnitude.

%}

% get frequency value
prompt1 = 'What frequency?';
frequency = input(prompt1);

% solve for omega
omega = 2 .* pi .* frequency;

% get T2 value
prompt2 = 'What T2?';
T2 = input(prompt2);

Mo = 1;

t = 0:1:100;

figure()


% plot real axis
subplot(311)
y =Mo .* exp(-t/T2) .* cos(omega *  t/1000);
plot(t,y)
title('Real Axis');
xlabel('Time (msecs)');
ylabel('Magnetization');
grid on

% plot imaginary axis
subplot(312)
z = Mo .* exp(-t/T2) .* sin(omega *  t/1000);
plot(t, z)
grid on


% plot magnitude
subplot(313)
Mt = sqrt(y.^2 + z.^2);
plot(t, Mt)
title('Magnitude');
xlabel('Time (msecs)');
ylabel('Magnitude');
grid on

File: Python MRI/T2PulseAcquire.py
"""

This program plots the results
of a T2 pulse acquire on the real axis and imaginary axis,
along with the magnitude.

"""


import numpy as np
import matplotlib.pyplot as plt

#get frequency value and solve for omega
frequency = input('Frequency: ')
omega =  2 *  np.pi * int(frequency)

#get T2 value
T2 = int(input('T2: '))

Mo = 1

t = np.arange(100)

plt.figure(1)

#graph real axis
plt.subplot(311)
plt.title('Real Axis')
plt.xlabel('Time (msecs)')
plt.ylabel('Magenetization')
plt.xlim(0,100)
plt.ylim(-1,1)
plt.grid(True)
Mx =  Mo * np.exp(-t/T2) * np.cos(omega * t/1000 )
plt.plot(t, Mx)

#graph imaginary axis
plt.subplot(312)
plt.title('Imaginary axis')
plt.xlabel('Time (msecs)')
plt.ylabel('Magenetization')
plt.xlim(0,100)
plt.ylim(-1,1)
plt.grid(True)
My = Mo * np.exp(-t/T2) * (np.sin(omega * t/1000))
plt.plot(t, My)


#graph magnitude
plt.subplot(313)
plt.title('Magnitude')
plt.xlabel('Time')  
plt.ylabel('Magnitude')
plt.xlim(0,100)
plt.ylim(0,1)
plt.grid(True)
Mt = np.sqrt(np.square(Mx) + np.square(My))
plt.plot(t, Mt)



#show graphs
plt.tight_layout()
plt.show()



